game.cpp

```diff
@@ -2183,6 +2205,32 @@ void test_game() //!OCLINT tests may be many
     }
   }
   #endif
+
+  #define FIX_ISSUE_542
+  #ifdef FIX_ISSUE_542
+  {
+      // Can update the timer of the scoring board
+      {
+          game g;
+          auto time_before = g.get_scoring_board().get_timer();
+          // update_timer() is within game::tick()
+          g.tick();
+          auto time_after = g.get_scoring_board().get_timer();
+          assert(time_after > time_before);
+      }
+
```

Comment:
Sure, I'll create more issues later, it's just that many ideas came up when I was writing the class. Our team is trying to work in the sprint mode, so we implement things really fast and are too eager to achieve the goal ASAP, sometimes we forget to break a big issue to small ones and attempt to fix a lot of things in only one PR

Reply:
Yeah, I understand the eagerness :)
Writing new issues + tests for all ideas and intermediate steps can be quite a speed bump to getting the code written indeed. But in my experience this method does speed things up in the long run, because
1. It makes it possible for other people to join and pick up the problem where you left it, thus multiplying your speed by the nb of contributors ;)
2. Debugging becomes much, much easier and faster if you work in small, self-contained issues.

Reply:
Just in case: no need to delete anything from the code! 
Just make new issues describing what should be in the class, refer to them in the tests you wrote, and maybe even close the issues if they're already solved.